### ISSUE TITLE
Default coloring for list does not work on Windows 7

### ISSUE THREAD

**Issue by robertvanbregt** [bram85/topydo#32]
When I run a `list` action, the color coding does not work. Instead of colors, I can see the equivalent characters, for example

<pre>
←[0m| 30| 2015-05-19 Buy milk ←[31m+Groceries←[0m←[0m←[0m←[31m@errands←[0m ←[31m
</pre>

**Comment by robertvanbregt** [comment]
A solution would be to add a flag like `-p` or `--plain` to remove any formatting in the output?

**Comment by bram85** [comment]
As a work-around, you can disable colors with a [configuration](https://github.com/bram85/topydo/wiki/Configuration) option:

```
[topydo]
colors = 0
```

**Comment by mruwek** [comment]
@robertvanbregt 

You can also try to install [ansicon](https://github.com/adoxa/ansicon).

@bram85
I've investigated [colorama](https://github.com/tartley/colorama) and [termcolor](https://pypi.python.org/pypi/termcolor), but those would implement only basic 16 colors.

**Comment by robertvanbregt** [comment]
The colors switch will do.

**Comment by bram85** [comment]
@mruwek: also I would like to keep topydo light on mandatory requirements, so I'm not so much in favor at this moment to create such a dependency just for the colors.

Still, colors should work for Windows.

**Comment by mruwek** [comment]
> Still, colors should work for Windows.

It's either colorama then or our own lib for translating ANSI codes to windoze `SetConsoleTextAttribute` calls.

**Comment by mruwek** [comment]
Or this: https://code.google.com/p/colorconsole/source/browse/colorconsole/win.py

**Comment by bram85** [comment]
@robertvanbregt Which console do you use where this color sequences appear in the output? Is it still an issue in the `master` branch?

**Comment by robertvanbregt** [comment]
I am using topydo 0.3.2 on my fairly locked down work PC. I run Win 7 SP
1 and launch a console from the Start > Run > Cmd prompt.

I have no git here, so it requires some effort to download and run the
master branch version. When I have time, I will give it a shot.

--
Groet, Regards,
Robert van Bregt
 
 
Op Wo jun 3 2015, om 07:59 schreef Bram Schoenmakers:
> @robertvanbregt[1] Which console do you use where this color sequences appear in the output? Is it still an issue in the master branch?


> —
> Reply to this email directly or view it on GitHub[2]. 

>  
 


Links:

  1. https://github.com/robertvanbregt
  2. https://github.com/bram85/topydo/issues/32#issuecomment-108202953

**Comment by bram85** [comment]
OK, I'm basically in the same boat, and I didn't see anything strange with cmd.exe when using the master branch

By the way, you don't need Git installed, with Github you can download the [latest master](https://github.com/bram85/topydo/archive/master.zip) from the site as a ZIP.

**Comment by mruwek** [comment]
> Is it still an issue in the master branch?

I don't have any Windows machine, but I'm quite sure it still doesn't work, because Windows doesn't support ANSI codes.

**Comment by bram85** [comment]
OK, never mind, I see it too now.

Note that cygwin may be a workaround that works for you as well. In my locked down environment I was able to install it (make sure to run the installer with the `-B` flag to omit the administrator prompt).

**Comment by robertvanbregt** [comment]
I have already installed a portable cygwin, for use with the original
todo.sh scripts. I found this too complex to use and decided to switch
to a native command line and Python. As stated in an earlier comment I
can live with a monochrome display (I have set `color=0` in my config to
eliminated escaped output.

--
Groet, Regards,
Robert van Bregt
 
 
Op Wo jun 3 2015, om 11:43 schreef Bram Schoenmakers:
> OK, never mind, I see it too now.


> Note that cygwin may be a workaround that works for you as well. In my locked down environment I was able to install it (make sure to run the installer with the -B flag to omit the administrator prompt).


> —
> Reply to this email directly or view it on GitHub[1]. 

>  
 


Links:

  1. https://github.com/bram85/topydo/issues/32#issuecomment-108273671
